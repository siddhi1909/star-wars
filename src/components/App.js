// IMPORT EXTERNAL LIBRARIES/MODULES
import React from 'react';
import {Provider} from "react-redux";
import {createBrowserHistory} from 'history';
import {
    Router,
    Switch,
    Route
} from "react-router-dom";
// IMPORT API & ROUTE ACTIONS
import './../index.css';
import Dashboard from './Dashboard';
import LoginComponent from './LoginComponent';
import PrivateRoute from '../routes/PrivateRoute';
import store from '../redux/store';

export default function App() {
    const Notfound = () => <h1> 404 Not found</h1>
    const history = createBrowserHistory();
    return (
        <Provider store={store}>
            <Router basename="/" history={history}>
                <Switch>
                    <Route exact path="/" component={LoginComponent}/>
                    <PrivateRoute exact path="/dashboard" component={Dashboard}/>
                    <Route path="*" component={Notfound}/>
                </Switch>
            </Router>
        </Provider>
    );
}