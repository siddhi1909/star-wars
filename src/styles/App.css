/*******Loader Style ********/
.logo {
    height: 20vmin;
    pointer-events: none;
}

.loader-custom {
    height: 150px;
    width: 150px;
    pointer-events: none;
    margin: auto;
}

@media (prefers-reduced-motion: no-preference) {
    .loader-custom {
        animation: loader-custom-spin infinite 20s linear;
    }
}

@keyframes loader-custom-spin {
    from {
        transform: rotate(0deg);
    }
    to {
        transform: rotate(360deg);
    }
}

.loader-outer {
    margin: 250px 0px;
}
/*******Loader Style ********/

.App-header {
    background-color: #282c34;
    min-height: 40vh;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    font-size: calc(10px + 2vmin);
    color: white;
}

.form-box {
    margin-top: 10px;
}

.padding-10-0{
    padding: 10px 0px;
}
.planet-title {
    font-size: 0.8rem;
    padding: 0px 5px;
}

.planet-outer {
    padding: 15px 0px;
    text-align: left;
    white-space: nowrap;
}

.hide-overflow {
    overflow: hidden
}

.user-dashboard {
    padding-right: 5px;
    margin: auto 0px;
    vertical-align: middle;
}

.wrapper {
    padding: 1.25rem;
}

.padding-top-bottom-10 {
    padding: 10px 0px;
}

.outer-div {
    padding: 1rem 0px;
}

/*******Pagination ********/
.custom-pagination-class.page-link.active {
    z-index: 1;
    color: #fff;
    background-color: #007bff;
    border-color: #007bff;
}
/*******Pagination ********/

/*****Toast **********/
.toast {
    align-items: center;
    border-radius: 25px;
    bottom: 0px;
    display: flex;
    min-height: 50px;
    max-height: 50px;
    justify-content: space-between;
    left:50%;
    padding: 0 20px;
    position: fixed;
    transform: translateX(-50%) translateY(150%);
    transition: transform .35s ease;
    z-index: 9;
}
.toast.visible {
    transform: translateX(-50%) translateY(-50%);
}

.toast.success {
    background-color: #16a085;
}

.toast.danger {
    background-color: #c0392b;
}
.toast.warning {
    background-color: #F38211;
}
.toast figure {
    height: 35px;
    margin: 0 15px 0 0;
    opacity: .9;
    width: 35px;
}
.toast figure img {
    height: 100%;
    width: 100%;
}
.toast p {
    color: #fff;
    font-family: 'Noto Sans';
    font-size: 14px;
    white-space: nowrap;
    margin-bottom: 0px;
}
/*****Toast **********/