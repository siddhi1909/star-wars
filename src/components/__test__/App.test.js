import React from 'react';
import renderer from 'react-test-renderer';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from '../App';

configure({ adapter: new Adapter() });

describe('>>>A P P --- Render REACT COMPONENTS',()=>{
    let wrapper

    beforeEach(()=>{
        wrapper = mount(<App/>)

    })

    it('+++ render the App component', () => {
        expect(wrapper.length).toEqual(1)
    });

    it('+++ < App /> renders correctly', () => {
        const renderedValue =  renderer.create(<App/>).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });

});