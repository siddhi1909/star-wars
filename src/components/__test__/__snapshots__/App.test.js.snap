// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>A P P --- Render REACT COMPONENTS +++ < App /> renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <img
      alt="logo"
      className="logo"
      src="star-wars-logo.png"
    />
  </div>
  <div
    className="container"
  >
    <form
      className="form-box"
      id="login"
      onSubmit={[Function]}
    >
      
      <div
        className="form-group"
      >
        <label
          htmlFor="username"
        >
          Username
        </label>
        <input
          className="form-control"
          id="username"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          required={true}
          type="text"
          value=""
        />
        <div
          className="invalid-feedback"
        >
          Please enter a valid username.
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="form-control"
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
        <div
          className="invalid-feedback"
        >
          Please enter your password to continue.
        </div>
      </div>
      <button
        className="btn btn-primary"
        type="submit"
      >
        Sign in
      </button>
    </form>
  </div>
</div>
`;

exports[`>>>A P P --- Render REACT COMPONENTS +++ < LoginComponent /> renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <img
      alt="logo"
      className="logo"
      src="star-wars-logo.png"
    />
  </div>
  <div
    className="container"
  >
    <form
      className="form-box"
      id="login"
      onSubmit={[Function]}
    >
      
      <div
        className="form-group"
      >
        <label
          htmlFor="username"
        >
          Username
        </label>
        <input
          className="form-control"
          id="username"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          required={true}
          type="text"
          value=""
        />
        <div
          className="invalid-feedback"
        >
          Please enter a valid username.
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="form-control"
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
        <div
          className="invalid-feedback"
        >
          Please enter your password to continue.
        </div>
      </div>
      <button
        className="btn btn-primary"
        type="submit"
      >
        Sign in
      </button>
    </form>
  </div>
</div>
`;
